realm/realm-library/src/androidTest/java/io/realm/DynamicRealmTests.java

```diff
@@ -402,6 +403,41 @@ private void populateForDistinct(DynamicRealm realm, long numberOfBlocks, long n
         realm.commitTransaction();
     }
 
+    private void populateForDistinctInvalidTypesLinked(DynamicRealm realm) {
+        realm.beginTransaction();
+        DynamicRealmObject notEmpty = realm.createObject(AllJavaTypes.CLASS_NAME);
+        notEmpty.setBlob(AllJavaTypes.FIELD_BINARY, new byte[]{1, 2, 3});
+        notEmpty.setObject(AllJavaTypes.FIELD_OBJECT, notEmpty);
+        notEmpty.setList(AllJavaTypes.FIELD_LIST, new RealmList<DynamicRealmObject>(notEmpty));
+        realm.commitTransaction();
+    }
+
+    /*
+     * Fields order test for Chained or Multi-Arguments Distinct()
+     *
+     * The idea is to interweave different values in 2's multiplier and 3's multiplier in a way that
+     * the outcome is different if the order of distinct* operations alternates. More numbers of
+     * fields can be constructed with the combination of multipliers in prime numbers such as 2, 3,
+     * and 5.
+     *
+     * An example is illustrated below.
+     *
+     * Object        : O1| O2| O3| O4| O5| O6
+     * indexedString : A | A | B | B | A | A
+     * indexedLong   : 1 | 1 | 1 | 2 | 2 | 2
```

Comment:
@beeender Thanks ;) The function is now in `TestHelper` as `populateForFieldsOrderSwitch()`.